Sources/NIOPosix/BaseSocketChannel.swift

```diff
@@ -140,12 +140,18 @@ private struct SocketChannelLifecycleManager {
                 pipeline.syncOperations.fireChannelUnregistered()
             }
 
+        // TODO: This is to support connect after bind for connected-mode datagram sockets. Might need a better solution for this.
```

Comment:
Does this TODO still apply?

Reply:
Yes, currently it is required. Without it, we get the following error.

```
NIOPosix/BaseSocketChannel.swift:166: Fatal error: illegal transition: state=activated, event=activate
2022-05-23 11:07:35.867416+0100 xctest[6470:11648282] NIOPosix/BaseSocketChannel.swift:166: Fatal error: illegal transition: state=activated, event=activate
```

I suppose, if we want to only allow this on `DatagramChannel` then we would need to change the implementation of `BaseSocketChannel.moveState(event:)` (currently private) to `fatalError("must be overridden")` and provide different implementations in `BaseStreamSocket` and `DatagramChannel`.

WDYT?